service/upstart/google-cloud-sql-proxy.conf

```diff
@@ -0,0 +1,46 @@
+# google-cloud-sql-proxy
+#
+# A MySQL proxy service provided by Google
+#
+# This upstart script expects the binary under /usr/local/bin/google-cloud-sql-proxy
+# It will create the sockets at /var/run/cloudsql
+# You can name CloudSQL instances via the compute instance metadata attribute name "cloud-sql-proxy-instances" - if you do not, all CloudSQL instances will be made available
+#
+# This upstart script uses named pipes to allow logging stdout and stderr to syslog via logger
+
+author "fogs@users.noreply.github.com"
+description "Google Cloud SQL Proxy"
+
+start on runlevel [2345]
+stop on runlevel [!2345]
+
+respawn
+respawn limit 10 5
+umask 022
+
+pre-start script
+    test -d /var/run/cloudsql || mkdir -p -m0755 /var/run/cloudsql
```

Comment:
This should just be "mkdir -p -m0755 /var/run/cloudsql". If the folder already exists, nothing will happen and mkdir will return an exit status of 0. If the folder does not exist and cannot be created, only then will it fail.